addons/payment/models/account_payment.py

```diff
@@ -28,12 +28,13 @@ def _get_payment_chatter_link(self):
 
     @api.depends('partner_id.commercial_partner_id.child_ids')
     def _compute_related_partners(self):
+
         for p in self:
             p.related_partner_ids = (
                 p.partner_id
               | p.partner_id.commercial_partner_id
               | p.partner_id.commercial_partner_id.child_ids
-            )._origin
+            )._origin if p.state == 'draft' else False
```

Comment:
Wouldn't a `search` with a domain with a clause `child_of` fix this?
I think it would even reduce the number of queries done, even though this is an explicit `search` in a loop...

Reply:
Or a `sudo`

Reply:
> Or a sudo

That would be simpler, indeed. Originally I discussed with @KangOl adding `compute_sudo=True` to the field but it seemed more intrusive to do otherwise. Let me know what you prefer and I'll update the PR.

Reply:
> Wouldn't a search with a domain with a clause child_of fix this?

I could test that...

Reply:
Hello @william-andre 

I couldn't figure out how to do this with `search`: since `partner_id` belongs to another company we cannot get `partner_id.commercial_partner_id`.
I've pushed a version using `compute_sudo=True`.
If you believe we better use `search`, please let me know how should we do that.

Thanks again for you review!